browser/android/tab_features_android.cc

```diff
@@ -5,11 +5,22 @@
 
 #include "brave/browser/android/tab_features_android.h"
 
+#include <memory>
+
+#include "brave/browser/ai_chat/ai_chat_utils.h"
+#include "brave/browser/ai_chat/tab_data_web_contents_observer.h"
+#include "chrome/browser/android/tab_android.h"
 #include "chrome/browser/profiles/profile.h"
 #include "content/public/browser/web_contents.h"
 
 TabFeaturesAndroid::TabFeaturesAndroid(content::WebContents* web_contents,
                                        Profile* profile)
-    : TabFeaturesAndroid_Chromium(web_contents, profile) {}
+    : TabFeaturesAndroid_Chromium(web_contents, profile) {
+  if (ai_chat::IsAllowedForContext(profile)) {
```

Comment:
sorry, not sure if I missed this last time, but we should prefer using a null check on the aichat service `AIChatServiceFactory::GetForBrowserContext`
it would be good if we could get rid of `ai_chat::IsAllowedForContext` in a follow-up

Reply:
I'm also ok leaving this for now and just doing them all in a follow-up

Reply:
https://github.com/brave/brave-browser/issues/43969